tfx/components/example_gen/import_example_gen/executor.py

```diff
@@ -43,11 +46,26 @@ def _ImportSerializedRecord(  # pylint: disable=invalid-name
       that maps to input files with root directory given by input_base.
 
   Returns:
-    PCollection of records (tf.Example, tf.SequenceExample, or bytes).
+    PCollection of records (tf.Example, tf.SequenceExample, bytes or dictionaries).
   """
   input_base_uri = exec_properties[standard_component_specs.INPUT_BASE_KEY]
   input_split_pattern = os.path.join(input_base_uri, split_pattern)
   logging.info('Reading input TFRecord data %s.', input_split_pattern)
+  output_payload_format = exec_properties.get(
+    standard_component_specs.OUTPUT_DATA_FORMAT_KEY)
+
+  def _InferArrowSchema(file_pattern):
+    match_result = FileSystems.match([file_pattern])[0]
+    files_metadata = match_result.metadata_list[0]
```

Comment:
this is alway the first match?

Reply:
not really, it is not exposed as an utility (internal method of the tfxio)